reconcile/utils/gql.py

```diff
@@ -134,13 +135,7 @@ def query(self, query, variables=None, skip_validation=False):
             if forbidden_schemas:
                 raise GqlApiErrorForbiddenSchema(forbidden_schemas)
 
-        if "errors" in result:
-            raise GqlApiError(result["errors"])
-
-        if "data" not in result:
-            raise GqlApiError(("`data` field missing from GraphQL" "server response."))
-
-        return result["data"]
+        return result.get("data")
```

Comment:
It looks like if the `data` key didn't exist before, we'd raise a `GqlApiError`, but `get()` will return `None` instead if the key doesn't exist. Is there a specific reason to change this?

Reply:
The client library now raises AssertionError if data key doesn't exist. If you follow the trail of calls happening within you should see this block of code. I have added a comment when the exception is caught first and rethrown as `GqlApiError`

```
        assert (
            result.data is not None
        ), "Transport returned an ExecutionResult without data or errors"
```

Reply:
So, `data` is guaranteed to be in the dictionary by the time you get this far? It looks like you switched it back to the original dict access method.

You may already be aware of this, but you should take care when using `some_dict['key']` versus `some_dict.get('key')`. If the key doesn't exist, with the former you'll get a `KeyError` exceptions, whereas the latter will just return `None`. If a key should absolutely exist, then don't use `get()`. If you're going to use `get()`, then the method type hints should indicate that it can return `None` and that has to be handled everywhere.

Reply:
> So, `data` is guaranteed to be in the dictionary by the time you get this far? It looks like you switched it back to the original dict access method.
> 
Yes, based on the library code `data` is guaranteed to be there.
> You may already be aware of this, but you should take care when using `some_dict['key']` versus `some_dict.get('key')`. If the key doesn't exist, with the former you'll get a `KeyError` exceptions, whereas the latter will just return `None`. If a key should absolutely exist, then don't use `get()`. If you're going to use `get()`, then the method type hints should indicate that it can return `None` and that has to be handled everywhere.
Yep, learned this detail recently :) I reverted to previous access method just to minimize the change.